File: frontend/src/api/ai.js
import { apiClient } from './client';

export async function ask(prompt, context) {
  const { data } = await apiClient.post('/ai/ask', { prompt, context });
  return data;
}

export async function hint(taskId, studentAttempt) {
  const { data } = await apiClient.post('/ai/hint', {
    task_id: taskId,
    student_attempt: studentAttempt ?? null,
  });
  return data;
}

File: backend/app/schemas/ai.py
from typing import Optional

from pydantic import BaseModel, Field


class AskIn(BaseModel):
    prompt: str = Field(..., min_length=1, max_length=2000)
    context: Optional[str] = Field(None, max_length=4000)


class AskOut(BaseModel):
    answer: str
    blocked: bool = False
    reason: Optional[str] = None


class HintIn(BaseModel):
    task_id: int
    student_attempt: Optional[str] = None
